roles/grafana_agent/tasks/main.yml

```diff
@@ -0,0 +1,51 @@
+---
+- name: Include secrets
+  include_vars: "{{ secrets_path | mandatory }}/mimir_password.yml"
+  no_log: true
+  tags:
+    - always
+
+- name: "Import Grafana GPG key"
+  become: true
+  ansible.builtin.apt_key:
+    url: "https://apt.grafana.com/gpg.key"
+    state: "present"
+  when: ansible_pkg_mgr == "apt"
+
+- name: "Add Grafana's repository to APT sources list"
+  become: true
+  ansible.builtin.apt_repository:
+    repo: "deb https://apt.grafana.com stable main"
+    state: present
+  when: ansible_pkg_mgr == "apt"
+
+- name: "Add Grafana's repository to yum/dnf systems"
+  become: true
+  ansible.builtin.template:
```

Comment:
Several ceph-cm-ansible roles were written in 2015 before Ansible had a [`yum_repository` module](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/yum_repository_module.html), and I understand this was probably copied from that pattern.

 It's much better to use the `yum_repository` module than to write our own `.repo` file with `template`.

(One example of the complexity that goes away with `yum_module`: the `mode` parameter should not be `0655` for a `.repo` file. That would make the file executable.)